solc/CommandLineInterface.cpp

```diff
@@ -438,16 +438,16 @@ bool CommandLineInterface::readInputFilesAndConfigureRemappings()
 					continue;
 				}
 
-				m_sourceCodes[infile.string()] = dev::readFileAsString(infile.string());
-				path = boost::filesystem::canonical(infile).string();
+				m_sourceCodes[infile.generic_string()] = dev::readFileAsString(infile.generic_string());
+				path = boost::filesystem::canonical(infile).generic_string();
```

Comment:
Perhaps it would be better to avoid the conversion to string here and directly add `boost::filesystem::canonical(infile).remove_filename()` to the allowed directories.